### PR TITLE
Attempt to fix intermittent failures in E2E tests

### PR DESCRIPTION
**Pull request by acoates-ms**
Not ideal, but nor is us having such a high failure rate on the current E2E tests.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5040)

### CODE REVIEW

File: packages/E2ETest/wdio/test/testInput.spec.ts [hunk 1]
```diff
@@ -26,6 +26,16 @@ describe('First', () => {
   it('Type abc on TextInput', () => {
     TextInputTestPage.clearAndTypeOnTextInput('abc');
     assert.equal(TextInputTestPage.getTextInputText(), 'abc');
+
+    // We should really develop a proper mechanism of waiting for idle, or detecting when the JS + bridge has had a chance of completing
+    // But this test is causing a large itermittent failure rate on CI, so adding this polling timeout for now.
+    // eslint-disable-next-line no-undef
+    browser.waitUntil(
+      () => TextInputTestPage.getTextInputPrev2Text().includes('onKeyPress'),
+      1000,
```
Comment (asklar) [inline packages/E2ETest/wdio/test/testInput.spec.ts:35]: if this is all it takes, great, but do we not need to wait in other parts of the test? IIRC I was seeing failures happening afterwards, when sending keystrokes. Maybe we need to add a delay on some of the test utility functions instead?

**Comment by JunielKatarn** [comment]
Out of curiosity, would that fix this kind of failure?
https://dev.azure.com/ms/react-native-windows/_build/results?buildId=83662&view=logs&jobId=8591e7ce-b1d3-5606-6030-d05adc61ff18&j=f83d74c5-64ae-5de5-afdb-0302ed3e4ae8&t=4895a795-b47f-5aa3-347f-f8e9bc731f50

The corresponding PR doesn't touch any E2E-related sources, but that job is still failing every single time.

**Comment by asklar** [comment]
> Out of curiosity, would that fix this kind of failure?
> https://dev.azure.com/ms/react-native-windows/_build/results?buildId=83662&view=logs&jobId=8591e7ce-b1d3-5606-6030-d05adc61ff18&j=f83d74c5-64ae-5de5-afdb-0302ed3e4ae8&t=4895a795-b47f-5aa3-347f-f8e9bc731f50
> 
> The corresponding PR doesn't touch any E2E-related sources, but that job is still failing every single time.

no, that's a different problem

File: packages/E2ETest/wdio/test/testInput.spec.ts [hunk 2]
```diff
@@ -26,6 +26,16 @@ describe('First', () => {
   it('Type abc on TextInput', () => {
     TextInputTestPage.clearAndTypeOnTextInput('abc');
     assert.equal(TextInputTestPage.getTextInputText(), 'abc');
+
+    // We should really develop a proper mechanism of waiting for idle, or detecting when the JS + bridge has had a chance of completing
+    // But this test is causing a large itermittent failure rate on CI, so adding this polling timeout for now.
```
Comment (vmoroz) [inline packages/E2ETest/wdio/test/testInput.spec.ts:31]: >itermittent [](start = 40, length = 11)

spelling

**Comment by dannyvv** [comment]
Perhaps as a backup to make things even more robust we could add azure devops as a "backup" feed by configuring multiple registries (npmjs first) but azdev as a backup since it seems npm is return 404's not 500's.

File: packages/E2ETest/wdio/test/testInput.spec.ts (hunk 1 above)
Comment (acoates-ms) [reply packages/E2ETest/wdio/test/testInput.spec.ts:35]: Lets try this for now and see how much it helps.  I consider this a bit of a hack, and would rather not apply it broadly.

**Pull request merged**